### PR TITLE
Fixing #10275 keyboard submit of adminhtml suggest form.

### PR DESCRIPTION
**Pull request by romainruaud**
This is a fix for #10275 

### Description

Pressing ENTER when navigating through admin suggest now work as expected.

### Fixed Issues (if relevant)

1. magento/magento2#10275 : Admin global search - submit by enter doesn't work 

### Manual testing scenarios

1. Go to Magento 2 admin panel
2. Click "Search" icon in header
3. Put some query (as example 7029) in search field, wait for autocomplete
4. Select "<your query>" in Products option with arrow keys on keyboard
5. Click enter button
6. You are redirected to the result page. Previously this was not working

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by romainruaud** [comment]
Hmmm the test which failed seems to have nothing related with my PR : 

```
1) Magento\Customer\Test\Unit\Model\AccountManagementTest::testInitiatePasswordResetEmailReminder

Expectation failed for method name is equal to <string:setRpTokenCreatedAt> when invoked zero or more times

Parameter 0 for invocation Magento\Customer\Model\Data\CustomerSecure::setRpTokenCreatedAt('2017-10-06 13:22:27') does not match expected value.

Failed asserting that two strings are equal.
```

What should I do now since I cannot restart the Travis build by myself @ishakhsuvarov  ?

**Comment by mzeis** [comment]
@romainruaud You can close and re-open the PR. This will trigger a new Travis build.

**Comment by romainruaud** [comment]
Travis timed out this time... I close/open to trigger again

File: lib/web/mage/backend/suggest.js [hunk 1]
```diff
@@ -245,6 +245,20 @@
 
                         case keyCode.ENTER:
                         case keyCode.NUMPAD_ENTER:
+                            suggestList = event.currentTarget.parentNode.getElementsByTagName('ul')[0];
```
Comment (ishakhsuvarov) [inline lib/web/mage/backend/suggest.js]: Is there any specific reason not to use jQuery for this purpose?

File: lib/web/mage/backend/suggest.js [hunk 2]
```diff
@@ -245,6 +245,20 @@
...
+                            hasSelectedItems = suggestList.getElementsByClassName('_active').length >= 0;
+
+                            if (hasSelectedItems) {
+                                selectedItem = $(suggestList.getElementsByClassName('_active')[0]);
+                                /* eslint-disable max-depth */
```
Comment (ishakhsuvarov) [inline lib/web/mage/backend/suggest.js]: We do not encourage suppressing warnings from linters. Please consider refactoring this part accordingly.

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]
Please check the code review suggestions and consider covering this behavior update with the test

**Comment by ishakhsuvarov** [comment]
Hi @romainruaud 
Do you have any updates regarding this PR? Any help wanted?

**Comment by romainruaud** [comment]
@ishakhsuvarov I was in holidays sorry.

I fix the changes ASAP

File: lib/web/mage/backend/suggest.js (hunk 1 above)
Comment (romainruaud) [reply lib/web/mage/backend/suggest.js]: @ishakhsuvarov you are right, I could have used jQuery, but vanillaJS does good also for things like this, and the rest of the file is retrieving elements via the same syntax, so I remained consistent.

File: lib/web/mage/backend/suggest.js (hunk 2 above)
Comment (romainruaud) [reply lib/web/mage/backend/suggest.js]: @ishakhsuvarov Yep, I will move this part into a dedicated function I guess.

**Comment by romainruaud** [comment]
@ishakhsuvarov done

**Comment by ishakhsuvarov** [comment]
@romainruaud Thank you!

**Review by ishakhsuvarov — APPROVED** [review]

**Pull request merged**